### PR TITLE
adding export for available positions. TM-1101

### PR DESCRIPTION
**Pull request by joelhanson**

### CODE REVIEW

**Comment by mjoyce91** [comment]
I'll want to dig into the transformations to make sure every thing is formatted correctly.

**Comment by mjoyce91** [comment]
@joelhanson I think there was a bad merge somewhere. Getting this error when I try to start the app:

```
NameError: name 'APIView' is not defined
```

**Comment by joelhanson** [comment]
@mjoyce91 There was a merge here I missed. Updating.

File: talentmap_api/fsbid/services/available_positions.py [hunk 1]
```diff
@@ -47,6 +50,67 @@ def get_available_positions_count(query, jwt_token, host=None):
...
+        smart_str(u"Posted Date"),
+        smart_str(u"Status Code"),
+    ])
+
+    for record in data:
+        writer.writerow([
+            smart_str(record["position"]["title"]),
+            smart_str(record["position"]["position_number"]),
+            smart_str(record["position"]["skill"]),
+            smart_str(record["position"]["grade"]),
```
Comment (mjoyce91) [inline talentmap_api/fsbid/services/available_positions.py]: Grade needs the same conversion mentioned above

File: talentmap_api/fsbid/services/available_positions.py [hunk 2]
```diff
@@ -47,6 +50,67 @@ def get_available_positions_count(query, jwt_token, host=None):
...
+            smart_str(record["position"]["post"]["location"]["country"]),
+            smart_str(record["position"]["tour_of_duty"]),
+            smart_str(record["position"]["languages"]),
+            smart_str(record["position"]["post"]["has_service_needs_differential"]),
+            smart_str(record["position"]["post"]["differential_rate"]),
+            smart_str(record["position"]["post"]["danger_pay"]),
+            smart_str(record["ted"]),
+            smart_str(record["position"]["current_assignment"]["user"]),
+            smart_str(record["bidcycle"]["name"]),
+            smart_str(record["posted_date"]),
```
Comment (mjoyce91) [inline talentmap_api/fsbid/services/available_positions.py]: All dates should be converted to `M/D/YYYY`

File: talentmap_api/fsbid/services/available_positions.py [hunk 3]
```diff
@@ -42,13 +48,75 @@ def get_available_positions(query, jwt_token, host=None):
...
+    writer.writerow([
+        smart_str(u"Position"),
+        smart_str(u"Position Number"),
+        smart_str(u"Skill"),
+        smart_str(u"Grade"),
+        smart_str(u"Bureau"),
+        smart_str(u"Post City"),
+        smart_str(u"Post Country"),
+        smart_str(u"Tour of Duty"),
+        smart_str(u"Languages"),
```
Comment (mjoyce91) [inline talentmap_api/fsbid/services/available_positions.py]: Can we have any arrays like Language be converted to omit the brackets? For example, a position with no languages returns as `[]`.

File: talentmap_api/fsbid/services/available_positions.py [hunk 4]
```diff
@@ -42,13 +48,75 @@ def get_available_positions(query, jwt_token, host=None):
...
+        query,
+        convert_ap_query,
+        jwt_token,
+        fsbid_ap_to_talentmap_ap,
+        "/api/v1/fsbid/available_positions/",
+        host
+    )
+
+    response = HttpResponse(content_type='text/csv')
+    response['Content-Disposition'] = "attachment; filename=available_positions.csv"
```
Comment (mjoyce91) [inline talentmap_api/fsbid/services/available_positions.py]: Can we append the current timestamp to this file name?

File: talentmap_api/fsbid/services/available_positions.py [hunk 5]
```diff
@@ -104,7 +104,7 @@ def get_available_positions_csv(query, jwt_token, host=None):
             smart_str(record["position"]["post"]["location"]["city"]),
             smart_str(record["position"]["post"]["location"]["country"]),
             smart_str(record["position"]["tour_of_duty"]),
-            smart_str(record["position"]["languages"]),
+            smart_str(record["position"]["languages"]).strip('[]'),
```
Comment (mjoyce91) [inline talentmap_api/fsbid/services/available_positions.py]: Will this work if languages exist? For example, this wouldn't remove the brackets from the following:

`[Spanish, French]`

Comment (mjoyce91) [reply talentmap_api/fsbid/services/available_positions.py]: Python `strip` works differently than I thought, so this will work! :+1

**Pull request merged**